src/java.base/share/classes/java/lang/ClassValue.java

```diff
@@ -500,15 +500,15 @@ <T> Entry<T> finishEntry(ClassValue<T> classValue, Entry<T> e) {
         synchronized
         void removeEntry(ClassValue<?> classValue) {
             Entry<?> e = remove(classValue.identity);
-            if (e == null) {
-                // Uninitialized, and no pending calls to computeValue.  No change.
-            } else if (e.isPromise()) {
-                // State is uninitialized, with a pending call to finishEntry.
-                // Since remove is a no-op in such a state, keep the promise
-                // by putting it back into the map.
-                put(classValue.identity, e);
-            } else {
-                // In an initialized state.  Bump forward, and de-initialize.
+            // e == null: Uninitialized, and no pending calls to computeValue.
+            // remove didn't change anything.  No change.
```

Comment:
Please capitalize the sentence: "Remove didn't…"
Also "Remove and discard" instead of "remove discarded".
Perhaps (this is being very picky) "Inside finishEntry, the logic will retry when it discovers the promise is removed."

This is a good bug fix.  Thanks for finding it.

Reply:
`remove` refers to the `WeakHashMap.remove(identity)` call. Clarified by using `remove(identity)` instead of just "remove".